### ISSUE TITLE
multi-label classification

### ISSUE THREAD

**Issue by YangjiaqiDig** [sbelharbi/deep-wsl-histo-min-max-uncertainty#6]
Hey! Can I extend your method to multi-label segmentation? 
instead of single pos label, but with multiple pos object classes?

**Comment by sbelharbi** [comment]
hi,
it should be ok. here are some suggestions:
- use binary cross-entropy at the output for each class.
- instead only of 2 masks, one needs K + 1: K is the total number of possible classes. + 1 for the background.
- masks can be softmaxed instead of sigmoid as in the paper: a pixel can belong only to one class or background.
- for an image x: 
   - classifier response: max p_c(x*mask_c) for c in the image, min p_c'(x*mask_c') for c' not in the image. min p_z(x*mask_bg) for z in all classes. this last term will replace the term R(p-) in the paper.
   - size: max size(mask_c) for c in the image, max size(mask_bg), min size(mask_c') for c' not in the image.

sorry for the very late reply.
thanks
